### ISSUE TITLE
Wrong location for cache directory

### ISSUE THREAD

**Issue by jaydiablo** [mimmi20/browscap-js#7]
In browscap.js this line loads the cache dir if one isn't present:

```
if (typeof cacheDir === 'undefined') {
    cacheDir = __dirname + '/node_modules/browscap-json-cache-files/sources/';
}
```

However, in my situation, I've installed browscap-js using npm:

`npm install browscap-js --save`

Which gives me this directory structure:

```
- node_modules
    - browscap-js
    - browscap-js-cache
    - browscap-json-cache-files
    - charenc
    - crypt
    - is-buffer
    - md5
```

When I run the sample code in the Readme, I get the `Default Browser` output.  If I navigate to the `browscap-js` directory, run `npm install` there, then everything works (because a `node_modules` directory is installed in the `browscap-js` directory, and the cache files are installed there as well).

If I `console.log(cacheDir)` in browcap.js, I get this:

`/project_root/node_modules/browscap-js/node_modules/browscap-json-cache-files/sources/`

Which, of course, doesn't normally exist unless I run `npm install` in the `browscap-js` folder inside my main `node_modules` folder.

Ideally, `cacheDir` should be:

`/project_root/node_modules/browscap-json-cache-files/sources/`

Looking at Node's documentation for `__dirname` it seems this is always resolved to be local to the module it is called from: https://nodejs.org/docs/latest/api/globals.html#globals_dirname

I can get around this by specifying my own `cacheDir`, but this isn't really clear in the documentation or examples.

Node version: v7.2.0
NPM version: 3.10.9

**Comment by mimmi20** [comment]
@jaydiablo Could you look into PR #8 and check if that solution is working? I dont have a project at the moment that uses browscap-js.

**Comment by jaydiablo** [comment]
Unfortunately, no, that doesn't work in this case, as the CWD for the call to the project's index.js file isn't necessarily the project's root (where node_modules is).

I was able to come up with a solution though (aside from passing the dir to the Browscap constructor) that should work and be pretty portable.  It would require a change to the browscap-json-cache-files project as well though.

Using Node's `require.resolve` we can get the path to the `browscap-json-cache-files` module, and then append "/sources" to that, it would look something like this:

```javascript
cacheDir = require('path').dirname(require.resolve('browscap-json-cache-files')) + '/sources';
```

Sadly, that won't work just by modifying this project, as you'll get an exception:

>Error: Cannot find module 'browscap-json-cache-files'

This happens because the `browscap-json-cache-files` module has no "index.js" file (or an equivalent file that is the module's main entry point, like `browscap.js` in this module).

As a test, I added an "index.js" file with the following content to the module in my node_modules folder, and then that code works as expected.

```javascript
module.exports = function BrowscapJsonCacheFiles() {

};
```

I'm not certain if the index.js file actually has to contain anything in this case, since `require.resolve` doesn't actually require anything, but maybe it's safer to include an empty export like this?

Using this method should always return the cache-files directory regardless of where the code is called from, or included from, as long as that module is present in the project (which it should be since it's listed as a dependency for `browscap-js` in the package.json.

With that code in place (and the index.js file in the other module in place) I can get this to work with my app, and I can run `npm test` in this module's root (after `npm install` of course) and all works as expected.

**Comment by mimmi20** [comment]
@jaydiablo I have updated both projects. Could you check, if this working now?

**Comment by jaydiablo** [comment]
I pointed my package.json at your branch from #8 and removed the directory from the constructor and it appears that all works well.  I made sure that my node_modules folder was empty (so that the node_modules folder in the `browscap-js` folder was empty) before testing to be sure it wasn't still referencing the JSON files there.

I also confirmed that tests within the `browscap-js` package run after npm installing the dependencies there.

👍
